delivery_multi_destination/tests/test_delivery_multi_destination.py

```diff
@@ -0,0 +1,207 @@
+                            "fixed_price": 50,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "Test child 2",
+                            "product_id": cls.product_child_2.id,
+                            "sequence": 2,
+                            "country_ids": [(6, 0, cls.country_2.ids)],
+                            "state_ids": [(6, 0, cls.state.ids)],
+                            "zip_from": 30000,
+                            "zip_to": 39999,
+                            "delivery_type": "fixed",
+                            "fixed_price": 150,
+                        },
+                    ),
+                ],
+            }
+        )
+        cls.carrier_single = cls.carrier_multi.copy(
+            {
+                "name": "Test carrier single",
+                "destination_type": "one",
+                "child_ids": False,
+            }
+        )
+        cls.product = cls.env["product.product"].create(
+            {"name": "Test product", "type": "product"}
+        )
+        cls.pricelist = cls.env["product.pricelist"].create(
+            {
+                "name": "Test pricelist",
+                "item_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "applied_on": "3_global",
+                            "compute_price": "formula",
+                            "base": "list_price",
+                        },
+                    )
+                ],
+            }
+        )
+        cls.sale_order = cls.env["sale.order"].create(
+            {
+                "partner_id": cls.partner_1.id,
+                "picking_policy": "direct",
+                "pricelist_id": cls.pricelist.id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "Test",
+                            "product_id": cls.product.id,
+                            "product_uom_qty": 1,
+                            "price_unit": 1,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    def test_delivery_multi_destination(self):
+        order = self.sale_order
+        order.carrier_id = self.carrier_single.id
+        delivery_wizard = Form(
+            self.env["choose.delivery.carrier"].with_context(
+                {
+                    "default_order_id": order.id,
+                    "default_carrier_id": order.carrier_id.id,
+                }
+            )
+        )
+        choose_delivery_carrier = delivery_wizard.save()
+        choose_delivery_carrier.button_confirm()
+        sale_order_line = order.order_line.search(
+            [("name", "=", "Test carrier single")]
+        )
```

Comment:
You should search by `is_delivery` flag, and the search is not limited by the order, so it's searching globally:
```suggestion
        sale_order_line = order.order_line.filtered("is_delivery")
```
do the same on the rest